indico/modules/categories/client/js/calendar.js

```diff
@@ -9,18 +9,69 @@
 
 import {Calendar} from '@fullcalendar/core';
 import dayGridPlugin from '@fullcalendar/daygrid';
+import moment from 'moment';
+import React from 'react';
+import ReactDOM from 'react-dom';
 
+import CalendarSingleDatePicker from 'indico/react/components/dates/CalendarSingleDatePicker';
+import {Translate} from 'indico/react/i18n';
+import {injectModal} from 'indico/react/util';
 import {$T} from 'indico/utils/i18n';
 
+import CalendarLegend from './components/CalendarLegend';
+
 (function(global) {
+  let groupBy = 'category';
+  const filteredLegendElements = new Set();
+  let modalShown = false;
   global.setupCategoryCalendar = function setupCategoryCalendar(
-    containerSelector,
+    containerCalendarSelector,
+    containerLegendSelector,
     tz,
-    categoryURL
+    categoryURL,
+    categoryId
   ) {
     const cachedEvents = {};
-    const container = document.querySelector(containerSelector);
+    const container = document.querySelector(containerCalendarSelector);
     const calendar = new Calendar(container, {
+      headerToolbar: {
+        start: 'prev,next goToDate today',
+        center: 'title',
+        end: 'dayGridMonth,dayGridWeek,dayGridDay',
+      },
+      customButtons: {
+        goToDate: {
+          text: Translate.string('Go to...'),
+          icon: 'i-button icon-calendar',
+          click: async (_, element) => {
+            if (!modalShown) {
```

Comment:
`if (modalShown) {return}`

then you don't need to have all the code below indented

Reply:
also close the modal when clicking the button again

Reply:
It does close and and reopen again. I don't really find a good way to fix it, since the `onClose` handler is called before the button's `onclick` method.

A dirty hack that occurred to me would be to enclose the `modalShown = false` line inside a function that executes after a given timeout. What do you think?

Reply:
find a nicer fix for it (see my commit and explanation on matrix)